src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
     public function canFilterAnyColumns($gridField)
     {
         $list = $gridField->getList();
-
-        if (!$this->checkDataType($list)) {
+        if (!($list instanceof Filterable) || !$this->checkDataType($list)) {
             return false;
         }
-
-        $columns = $gridField->getColumns();
-        foreach ($columns as $columnField) {
-            $metadata = $gridField->getColumnMetadata($columnField);
-            $title = $metadata['title'];
-
-            if ($title && $list->canFilterBy($columnField)) {
-                return true;
+        $modelClass = $gridField->getModelClass();
+        $searchableFields = Config::inst()->get($modelClass, 'searchable_fields');
```

Comment:
Searchable fields should be fetched by calling `searchableFields()` on a singleton, i.e. `$modelClass::singleton()->searchableFields()`
See [SearchContext](https://github.com/silverstripe/silverstripe-framework/blob/a4929a171e0e0956ecff6f49b04a88f1ec9c26f9/src/ORM/Search/SearchContext.php#L164-L165) and [GridFieldAddExistingAutocompleter](https://github.com/silverstripe/silverstripe-framework/blob/a4929a171e0e0956ecff6f49b04a88f1ec9c26f9/src/Forms/GridField/GridFieldAddExistingAutocompleter.php#L340-L342) as examples of where this is already done.

Reply:
Updated